go-controller/pkg/ovn/ovn.go

```diff
@@ -501,13 +502,19 @@ func (oc *Controller) syncNodeGateway(node *kapi.Node, subnet *net.IPNet) error
 
 // WatchNodes starts the watching of node resource and calls
 // back the appropriate handler logic
-func (oc *Controller) WatchNodes(nodeSelector *metav1.LabelSelector) error {
+func (oc *Controller) WatchNodes() error {
 	var gatewaysFailed sync.Map
 	macAddressFailed := make(map[string]bool)
-	_, err := oc.watchFactory.AddFilteredNodeHandler(nodeSelector, cache.ResourceEventHandlerFuncs{
+	nodeSelector, _ := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
+		MatchLabels: map[string]string{kapi.LabelOSStable: "linux"},
+	})
+	_, err := oc.watchFactory.AddNodeHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			node := obj.(*kapi.Node)
 			logrus.Debugf("Added event for Node %q", node.Name)
+			if !nodeSelector.Matches(labels.Set(node.Labels)) {
```

Comment:
@JacobTanenbaum I would move this into addNode() to keep all that logic together and use the config option I talked about above. Then addNode() can return (nil, nil) when it ignores a node, and you can check that like:

```
if err != nil {
    <log error allocating node>
    return
} else if hostSubnet == nil {
    <log that we ignored this node>
    return
}
```